graphql/resolvers.go

```diff
@@ -1819,29 +1819,65 @@ func (r *mutationResolver) UnschedulePatchTasks(ctx context.Context, patchID str
 	return &patchID, nil
 }
 
-// We only allow scheduling undispatched base tasks for tasks that are failing on the current patch
+// ScheduleUndispatchedBaseTasks only allows scheduling undispatched base tasks for tasks that are failing on the current patch
 func (r *mutationResolver) ScheduleUndispatchedBaseTasks(ctx context.Context, patchID string) ([]*restModel.APITask, error) {
 	opts := data.TaskFilterOptions{
 		Statuses:     evergreen.TaskFailureStatuses,
 		BaseStatuses: []string{evergreen.TaskUndispatched},
 	}
 	tasks, _, err := r.sc.FindTasksByVersion(patchID, opts)
 	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch :%s ", err.Error()))
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch: %s ", err.Error()))
 	}
 
 	restartedTasks := []*restModel.APITask{}
+	generatedTasksToRestart := []*task.Task{}
+
 	for _, t := range tasks {
-		// Can ignore error while fetching tasks this could just mean the task didn't exist on the base commit
+		// If a task is a generator task or generated don't schedule it until we get all of the generated tasks we want to generate
+		if !t.GenerateTask && t.GeneratedBy == "" {
+			// We can ignore an error while fetching tasks because this could just mean the task didn't exist on the base commit.
+			baseTask, _ := t.FindTaskOnBaseCommit()
+			if baseTask != nil {
+				task, err := SetScheduled(ctx, r.sc, baseTask.Id, true)
+				if err != nil {
+					return nil, err
+				}
+				restartedTasks = append(restartedTasks, task)
+			}
+			// If a task is generated lets keep track of it so we can restart it later
+		} else if t.GeneratedBy != "" {
+			generatedTasksToRestart = append(generatedTasksToRestart, &t)
+
+		}
+	}
+
+	baseGeneratorTasksToSchedule := []*task.Task{}
+	// If a generated task does not have a base task find its generator and use its base task
+	// to generate it
+	for _, t := range generatedTasksToRestart {
 		baseTask, _ := t.FindTaskOnBaseCommit()
-		if baseTask != nil {
-			task, err := SetScheduled(ctx, r.sc, baseTask.Id, true)
+		if baseTask == nil {
+			generatorTask, err := task.FindByIdExecution(t.GeneratedBy, &t.Execution)
 			if err != nil {
-				return nil, err
+				return nil, InternalServerError.Send(ctx, fmt.Sprintf("Experienced an error trying to find the generator task: %s", err.Error()))
+			}
+			// Lets schedule the generated tasks on the base generator task
+			baseGeneratorTask, _ := generatorTask.FindTaskOnBaseCommit()
+			baseGeneratorTask.SetGeneratedTasksToStepback(t.BuildVariant, t.DisplayName)
+			// If we plan on scheduling a generator task already make sure we don't try to schedule it twice
+			i := sort.Search(len(baseGeneratorTasksToSchedule), func(i int) bool { return baseGeneratorTasksToSchedule[i].Id == generatorTask.Id })
```

Comment:
This is maybe too intelligent/not optimized; you can just use a map taskId -> true and then iterate through the map at 1875